packages/evolution-common/src/services/baseObjects/Segment.ts

```diff
@@ -0,0 +1,511 @@
+/*
+ * Copyright 2024, Polytechnique Montreal and contributors
+ *
+ * This file is licensed under the MIT License.
+ * License text available at https://opensource.org/licenses/MIT
+ */
+
+import { Optional } from '../../types/Optional.type';
+import { IValidatable, ValidatebleAttributes } from './IValidatable';
+import { Uuidable, UuidableAttributes } from './Uuidable';
+import { WeightableAttributes, Weight, validateWeights } from './Weight';
+import * as SAttr from './attributeTypes/SegmentAttributes';
+import { Junction, JunctionAttributes } from './Junction';
+import { Routing, RoutingAttributes } from './Routing';
+import { Result, createErrors, createOk } from '../../types/Result.type';
+import { ParamsValidatorUtils } from '../../utils/ParamsValidatorUtils';
+import { ConstructorUtils } from '../../utils/ConstructorUtils';
+
+export const segmentAttributes = [
+    '_weights',
+    '_isValid',
+    '_uuid',
+    'modeCategory',
+    'mode',
+    'modeOtherSpecify',
+    'departureDate',
+    'arrivalDate',
+    'departureTime',
+    'arrivalTime',
+    'driver',
+    'driverUuid',
+    'carType',
+    'vehicleOccupancy'
+];
+
+export const segmentAttributesWithComposedAttributes = [
+    ...segmentAttributes,
+    'origin',
```

Comment:
Shouldn't the origin and destination be part of the segment's main attribute and not an extended one?

Reply:
it is a composed object (Place) Extended means standrad attributes + composed + custom
composed attributes =  attributes + composed